src/ddmd/func.d

```diff
@@ -156,6 +156,61 @@ enum FUNCFLAGinlineScanned    = 0x20;   /// function has been scanned for inline
 enum FUNCFLAGinferScope       = 0x40;   /// infer 'scope' for parameters
 
 
+/****************************************************
+ * Determine whether an 'in' contract is declared inside
+ * the given function or any of its overrides.
+ * Params:
+ *      fd = the function to search
+ * Returns:
+ *      true    found an 'in' contract
+ *      false   didn't find one
+ */
+private bool needsFrequire(FuncDeclaration fd)
+{
+    if (fd.frequire)
+        return true;
+
+    for (size_t i = 0; i < fd.foverrides.dim; i++)
+    {
+        FuncDeclaration fdv = fd.foverrides[i];
+
+        if (fdv.frequire)
+            return true;
+
+        if (needsFrequire(fdv))
+            return true;
```

Comment:
test cases for the two red lines 181 and 208?

Reply:
OK, I have test cases for them - whether this is hit depends on order of declarations.
```
class C : B { override int foo() { return 1; } }
class B : A { override int foo() { return 2; } }
class A { int foo() out(res) { assert(res != 0); } body { return 0; } }
```
From the above, functions will be analysed in order of `C.foo`, `B.foo`, `A.foo`, in which case, recursive search is required (`B.foo` hasn't started semantic, and so does not yet have `fensure` set).